lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,45 @@
+Facter.add(:powershell_version) do
+  setcode do
+    if Puppet::Util::Platform.windows?
+      require 'win32/registry'
+      version = nil
+      begin
+        access_type = Win32::Registry::KEY_READ | 0x100
+        hive        = Win32::Registry::HKEY_LOCAL_MACHINE
+        hive.open('SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine', access_type) do |reg|
+          version = reg['PowerShellVersion']
+        end
+      rescue
+        version = nil
```

Comment:
This is because this fact is before a confine to a posh version, so it could be run against PowerShell v2, which will be in the `1` key and not the `3` key.